src/components/Auth/useAuthProvider.ts

```diff
@@ -58,49 +38,64 @@ enum LocalStorageKeys {
 }
 
 export const useAuthProvider = () => {
-  const { setSigner, clear } = useClient()
+  const { signer, setSigner, clear } = useClient()
   const [bearer, setBearer] = useState<string | null>(localStorage.getItem(LocalStorageKeys.AUTH_TOKEN))
   const login = useLogin({
     onSuccess: (data, variables) => {
-      console.log('logged in', data, variables)
       storeLogin(data)
     },
   })
   const register = useRegister({
     onSuccess: (data, variables) => {
-      console.log('registered', data, variables)
       storeLogin(data)
     },
   })
 
   const isAuthenticated = useMemo(() => !!bearer, [bearer])
 
-  // todo: implement this
-  // check if the token is still valid
-  // useEffect(() => {
-  //   if (!bearer) return
-  //
-  //   bearedFetch(`${appUrl}/auth/check`)
-  //     // Invalid token or expired, so logout
-  //     .then(async (response) => {
-  //       if (response.status !== 200) {
-  //         logout()
-  //       }
-  //     })
-  //     // network errors or other issues
-  //     .catch(() => {
-  //       logout()
-  //     })
-  // }, [])
+  const bearedFetch = useCallback(
```

Comment:
@selankon this seems like a generic function, as the one implemented in `api.ts`. 
Is it going to be used by future calls as well? If yes, it should be treated as `api.ts`, proably, outside the components folder